ios/FluentUI/Navigation/UIBarButtonItem+BadgeValue.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc public extension UIBarButtonItem {
+    private struct AssociatedKeys {
+        static var badgeValue: String = "badgeValue"
+    }
+
+    static let badgeValueDidChangeNotification = NSNotification.Name(rawValue: "UIBarButtonItemBadgeValueDidChangeNotification")
+
+    /// The badge value will be displayed in a red oval above the UIBarButtonItem.
+    /// Set the badge value to nil to hide the red oval.
+    @objc var badgeValue: String? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKeys.badgeValue) as? String ?? nil
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.badgeValue, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+            NotificationCenter.default.post(name: UIBarButtonItem.badgeValueDidChangeNotification, object: self)
+        }
+    }
+
+    /// Convenience method to set the badge value to a number.
+    /// If the number is zero, the badge value will be hidden.
+    @objc func setBadgeNumber(_ number: UInt) {
```

Comment:
do we need the badgenumer?

Reply:
Not so required but IMO it would be better to give such an API to the developers using the UIBarButtonItem just like we are giving in case of TabBarItem.